File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLPhoneCallDiscardReasonMissed.java
package com.github.badoualy.telegram.tl.api;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLPhoneCallDiscardReasonMissed extends TLAbsPhoneCallDiscardReason {

    public static final int CONSTRUCTOR_ID = 0x85e42301;

    private final String _constructor = "phoneCallDiscardReasonMissed#85e42301";

    public TLPhoneCallDiscardReasonMissed() {
    }

    @Override
    public String toString() {
        return _constructor;
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLPhoneCallDiscardReasonMissed.kt
package com.github.badoualy.telegram.tl.api

/**
 * phoneCallDiscardReasonMissed#85e42301
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLPhoneCallDiscardReasonMissed : TLAbsPhoneCallDiscardReason() {
    private val _constructor: String = "phoneCallDiscardReasonMissed#85e42301"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLPhoneCallDiscardReasonMissed) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x85e42301.toInt()
    }
}
